src/core/utils/server.js

```diff
@@ -0,0 +1,63 @@
+/* @flow */
+import path from 'path';
+
+import fs from 'fs-extra';
+import config from 'config';
+import type { $Request, $Response } from 'express';
+
+import log from 'core/logger';
+
+const PREFIX = 'enableFeature';
+
+export const getFeatureFlagName = (featureFlagKey: string): string => {
+  return featureFlagKey.replace(PREFIX, '');
+};
+
+type ExpressHandler = (req: $Request, res: $Response) => void;
+
+type ViewFrontendVersionHandlerParams = {|
+  _config?: typeof config,
+  _log?: typeof log,
+  versionFilename?: string,
+|};
+
+export const viewFrontendVersionHandler = ({
+  _config = config,
+  _log = log,
+  // This is a magic file that gets written by deployment scripts.
+  versionFilename = 'version.json',
+}: ViewFrontendVersionHandlerParams = {}): ExpressHandler => {
+  const version = path.join(_config.get('basePath'), versionFilename);
+
+  const featureFlags = Object.keys(_config)
+    .filter((key) => {
+      return key.startsWith(PREFIX);
+    })
+    .reduce((map, key) => {
+      return {
+        ...map,
+        [getFeatureFlagName(key)]: _config.get(key),
+      };
+    }, {});
+
+  const experiments = _config.get('experiments');
+
+  return (req: $Request, res: $Response) => {
+    fs.stat(version, async (error) => {
+      if (error) {
+        _log.error(
+          `Could not stat version file ${version}: ${error.toString()}`,
+        );
+        res.sendStatus(415);
+      } else {
+        const versionJson = await fs.readJson(version);
+
+        res.json({
+          ...versionJson,
+          experiments,
+          feature_flags: featureFlags,
```

Comment:
`featureFlags` (instead of `feature_flags`) would be more consistent with our code base.

Reply:
This endpoint will be used by another app in the near future, so making it pretty could have been nice. The API we consume uses snake case for the JSON properties, so I just used that here too.

Reply:
I think the backend (which will eventually expose feature flags) will use snake case, so that would be nice to be consistent.